core/src/main/scala/io/snappydata/Literals.scala

```diff
@@ -121,6 +121,25 @@ object Constant {
   val CHANGEABLE_JAR_NAME = "SNAPPY_CHANGEABLE_JAR_NAME"
 
   val RESERVOIR_AS_REGION = "spark.sql.aqp.reservoirAsRegion"
+
+  // -10 in sequence will mean all arguments, -1 will mean all odd argument and
+  // -2 will mean all even arguments.
+  // @TODO check whether function like named_struct, ntile etc. can ever
+  // come in the where clause of a query. Right now Tokenization is done
+  // for constants in where clause only.
+  val FOLDABLE_FUNCTIONS: Map[String, Seq[Int]] = Map("ROUND" -> Seq(1),
+    "BROUND" -> Seq(1), "PERCENTILE" -> Seq(0, 1), "STACK" -> Seq(-10),
+    "NTILE" -> Seq(0), "STR_TO_MAP" -> Seq(1, 2), "NAMED_STRUCT" -> Seq(-1),
+    "REFLECT" -> Seq(1, 2), "JAVA_METHOD" -> Seq(1, 2), "XPATH" -> Seq(2),
+    "XPATH_BOOLEAN" -> Seq(2), "XPATH_DOUBLE" -> Seq(2),
+    "XPATH_NUMBER" -> Seq(2), "XPATH_FLOAT" -> Seq(2),
+    "XPATH_INT" -> Seq(2), "XPATH_LONG" -> Seq(2),
+    "XPATH_SHORT" -> Seq(2), "XPATH_STRING" -> Seq(2),
+    "PERCENTILE_APPROX" -> Seq(0, 1, 2), "APPROX_PERCENTILE" -> Seq(0, 1, 2),
+    "TRANSLATE" -> Seq(1, 2), "UNIX_TIMESTAMP" -> Seq(0),
+    "TO_UNIX_TIMESTAMP" -> Seq(0), "FROM_UNIX_TIMESTAMP" -> Seq(0),
+    "TO_UTC_TIMESTAMP" -> Seq(0), "FROM_UTC_TIMESTAMP" -> Seq(0),
+    "TRUNC" -> Seq(0, 1), "NEXT_DAY" -> Seq(0, 1))
```

Comment:
@kneeraj Yes, considering that I said 2nd argument because its the format which is a constant and not the value itself e.g. `SELECT UNIX_TIMESTAMP('2016-04-08', 'yyyy-MM-dd')`
See the class comment of UnixTimestamp. Looks like Spark allows for same function to have different number of arguments. This function can take 0 arguments, 1 argument or 2 arguments, but need to disable tokenization only for the 2nd argument when specified.